services/structureddate/structureddate/src/main/java/org/collectionspace/services/structureddate/DateUtils.java

```diff
@@ -1120,16 +1120,34 @@ public static Date getLatestAfterDate(Date startDate, Date endDate) {
 			return currentDate;
 		}
 
-		MutableDateTime currentDateTime = convertToDateTime(currentDate);
-		MutableDateTime endDateTime = convertToDateTime(endDate);
-		
-		int comparisonResult = currentDateTime.compareTo(endDateTime);
+		int comparisonResult = compareDates(currentDate, endDate);
 		if (comparisonResult == 1 || comparisonResult == 0) {
 			return currentDate;
 		}
 		return null;
 	}
 
+	/**
+	 * Wrapper function for MutableDateTime's comparator.
+	 * @param startDate The first date in the range
+	 * @param endDate   The last date in the range
+	 * @return          -2 if either are null, -1 if startDate is before,
+	 * 					0 if they are equal, 1 if startDate is after endDate
+	 */
+	public static int compareDates(Date startDate, Date endDate) {
```

Comment:
This is really going to confuse people. Normally a comparator returns a negative number if arg 1 < arg2, zero if arg1 = arg2, or a positive number if arg1 > arg2. No one is going to expect a specific negative number to mean that one of the args was null. If you want to handle nulls you can treat them as ordered before non-nulls. If you don't want to handle nulls, you can throw an exception. If the caller wants to treat nulls some other way,  the caller should check for nulls and do what it wants before (or instead of) calling this function.